lec16-data-viz.Rmd

```diff
@@ -1,4 +1,97 @@
 ---
-title: "More on data visualizations (plus project work)"
+title: "More on data visualizations and the git workflow (plus project work)"
+author: Joel Ostblom
 ---
 
+First, let's create a chat room for your repository. We will use gitter, since
+it has good integrations with GitHub. Got to http://gitter.im/, login with your
+GitHub account and click the plus sign to create a new chat room. Browse to
+your repository, and add the members of your team. Since your repository is
+public, this chatroom will also be public (so you might not want to share
+personal information, and keep your discussions on topic, just as if this was
+a real-world work project).
+
+Discussions in the chat room are good to crystallize what tasks need to be
+tackled next. Once you have identified a specific task, open an issue about it
+on GitHub to track the progress of that particular tasks. You should create
+issues for your planned analyses, writing, literature review, etc. Then
```

Comment:
> Create an issue for problems that come up that need to be discussed with others. (if you want the problem in writing... you could just talk in person too).

Reply:
I think they should create issues for most tasks. That is how we will be able to judge their progress, and it will teach them to use GitHub. Exceptions would be minor changes, such as fixing spelling mistakes.

Reply:
Well, I don't think issues should be created if they are part of doing the assignment. Are they going to make an issue for "do correlation analysis", "write paragraph for methods", etc? That would simply impede progress. We don't even do that!

Reply:
Plus, it is called an "Issue" as in, a "problem" that needs to be addressed.

Reply:
Yes, I think they should use the issues for planning tasks like the ones specified. I think of issues as actionable items, and when they write down their plan, they will have tasks that needs to be addressed, I suggest that these are converted into issue for keeping track of progress.

You could also say that the "issue" that needs to be solved is that they have not done the analyses, written the section, etc. It's like using issues for feature requests. If they don't create issues for that, what do you have in mind that they will create issues for? Only when they run into troubles with their code?

Reply:
I personally just think that by forcing them to put all tasks down that it will slow their progress, since they will have to manage and create these issues. Since they are working in person in teams, they can just communicate what each of their tasks are with each other. Just think of your own research, or like dplyr or other open source projects. They (dplyr/tidyverse) don't create issues for tasks that their team is working on, they leave the issues for other non-team users to submit bug reports, enhancements, features, etc. 

What I think you want of them is for them to do some project management to track progress... but GitHub issues were not designed to be used for project management. I'd rather they focus on just getting the stuff done and writing out what they will do and then doing it.

And I was actually not really thinking that they would use issues much if at all. We might use the issues to get them to address something, but they won't need to.

Plus, we track their progress by looking at commits, PR, and in-person help/looking over.

Reply:
My recommendation would be to just say that if there is a problem or issue they need to deal with *eventually* but not right away, they can create an issue.

Reply:
I was just going to write that I want them to use issues for basic project management... =) I do try to use issues for as many tasks as possible, everything I cannot fix in 5 min basically (both for code and research). I have tried to keep these things in memory, but I always forget stuff that I don't do immediately and since I will be writing it down anyways, why not write it down as an issue that I can address?

I agree that there are better tools for project management, including GitHub addons and GitLab (anything with a Kanban board really), but as you say, it is unnecessary to go overboard here. But I think it is important to teach them how to plan and then adjust as a team when things do not go according to plan. Are you thinking that they will have the initial plan that is part of the assignment, and then deal with changes to that plan by talking (or updating the text file) instead of via issues?

 I was hoping that they will use issues a lot! Our opinions differ! Would you have time to meet ~30 min before class and discuss in person? ... or should I create an issue about it? (⊙_⊙)

Reply:
Yes, I can come ~30 min before hand :)

Reply:
Great, thank you!